parser-typechecker/src/Unison/DataDeclaration.hs

```diff
@@ -47,7 +47,11 @@ type ConstructorId = Int
 
 type DataDeclaration v = DataDeclaration' v ()
 
+data Modifier = Structural | Unique Text -- | Opaque (Set Reference)
```

Comment:
This is really the key change. All type declarations have a modifier, and then I went and fixed all the compile errors.